apps/admin-dashboard/app/routes/_dashboard.events.sync-airmeet-event.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  type ActionFunctionArgs,
+  json,
+  type LoaderFunctionArgs,
+  redirect,
+} from '@remix-run/node';
+import {
+  Form as RemixForm,
+  useActionData,
+  useNavigate,
+  useNavigation,
+} from '@remix-run/react';
+import { z } from 'zod';
+
+import {
+  Button,
+  Form,
+  getActionErrors,
+  Input,
+  Modal,
+  validateForm,
+} from '@oyster/ui';
+
+import { Route } from '../shared/constants';
+import { job } from '../shared/core.server';
+import {
+  commitSession,
+  ensureUserAuthenticated,
+  toast,
+} from '../shared/session.server';
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  await ensureUserAuthenticated(request);
+
+  return json({});
+}
+
+const SyncAirmeetEventFormData = z.object({
+  eventId: z.string().uuid('The ID must be a valid UUID.'),
+});
+
+export async function action({ request }: ActionFunctionArgs) {
+  const session = await ensureUserAuthenticated(request);
+
+  const form = await request.formData();
+
+  const { data, errors } = validateForm(
+    SyncAirmeetEventFormData,
+    Object.fromEntries(form)
+  );
+
+  if (!data) {
+    return json({
+      error: '',
+      errors,
+    });
+  }
+
+  job('event.sync', {
+    eventId: data.eventId,
```

Comment:
To answer your question about checking the type of `eventId`, the `data` object is validated (see the `validateForm` above in which we pass our schema), so we can just use `data.eventId`!